[P1] CMMS stats API returns 503 on staging (likely atlas-api timing)

## Summary
`GET /api/cmms/stats/` returned HTTP 503 during the 2026-05-19 staging audit. Atlas CMMS may have still been starting up.

## User impact
`/cmms` page shows degraded state — KPI tiles cannot load.

## Evidence
- Audit run: `docs/promo-screenshots/audit-2026-05-19-staging-report.json` (cmms row, mobile viewport)
- Screenshot: `2026-05-19_cmms_mobile.png`
- Console: `503 /api/cmms/stats/`

## Notes
- May be transient — atlas-api healthcheck timing during boot
- Needs **re-verification**: `curl http://127.0.0.1:4101/api/cmms/stats/` after stack settles
- If persistent: fix healthcheck dependency in `docker-compose.staging.yml` (Hub should wait on atlas-api ready, or proxy should retry)

## Source
- Route: `mira-hub/src/app/api/cmms/stats/route.ts`
- Health: `mira-hub/src/app/api/cmms/health/route.ts`

## Acceptance
- After full stack warm-up, `/api/cmms/stats/` returns 200
- Re-run audit; `cmms` row shows `failedRequests: []`
- If timing-related, document/fix the dependency

## Priority
P1 — could be transient; needs verification before fix scope is decided.

**Update (2026-05-19 mid-day):** Partial fix and root cause clarified.

What I cleared (the original symptom):
- Stale Liquibase changelog lock on stg-atlas-db was holding atlas-api in a crash-restart loop. Cleared via \`UPDATE databasechangeloglock SET locked=false\` after \`docker stop stg-atlas-api\` to avoid race re-acquisition, then started fresh.
- Liquibase finally completed all 762 changesets. Atlas-db schema is current.

What's still broken:
- atlas-api now crashes on init at \`Caused by: com.grash.exception.CustomException: Error configuring MinIO: Failed to connect to /165.245.138.91:4900\`.
- atlas-api's bean init calls the **public** MinIO URL (\`ATLAS_PUBLIC_MINIO_URL=http://165.245.138.91:4900\`), but from inside the Docker network the container can't loopback to its own host's public IP — Docker's default bridge has no hairpin NAT.
- Verified with \`wget\` from inside stg-atlas-api: \`stg-atlas-minio:9000\` returns 403 (reachable), \`165.245.138.91:4900\` times out (unreachable).

Real fix is one of:
1. Change \`ATLAS_PUBLIC_MINIO_URL\` to a host-reachable value from inside the network (e.g., \`host.docker.internal\` + Linux \`extra_hosts: host-gateway\` mapping in compose), OR
2. Patch atlas-api to use the internal \`MINIO_ENDPOINT\` for init checks instead of \`PUBLIC_MINIO_ENDPOINT\` — the public URL is only needed for browser-facing signed URLs.

Leaving as P1 — staging \`/cmms\` still shows 503 but no production impact.

--- a/mira-cmms/docker-compose.yml
+++ b/mira-cmms/docker-compose.yml
@@ -48,6 +48,13 @@ services:
     image: intelloop/atlas-cmms-backend@sha256:21c42377dd758dbc4a9aa1676e62b01a13cf8822b0ecd3019b274b53c2369b1e
     container_name: atlas-api
     restart: unless-stopped
+    # atlas-api's bean init connects to PUBLIC_MINIO_ENDPOINT at startup; Linux Docker
+    # bridge has no hairpin NAT, so a host-public-IP value crash-loops the container.
+    # Deploys that need to point the public URL at the host gateway can set
+    # ATLAS_PUBLIC_MINIO_URL=http://host.docker.internal:<port>. Mac/Windows already
+    # have host.docker.internal; Linux needs the explicit mapping.
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
     depends_on:
       atlas-db:
         condition: service_healthy

